lib/react/rails/railtie.rb

```diff
@@ -65,7 +65,7 @@ class Railtie < ::Rails::Railtie
       config.after_initialize do |app|
         # Server Rendering
         # Concat component_filenames together for server rendering
-        app.config.react.components_js = lambda {
+        app.config.react.components_js ||= lambda {
```

Comment:
@rearadmiral  `x ||= 10`, expands to `x = x || 10` . This is not guarded, and all three expressions in the latter are sequential, so x is not always guaranteed to be 10 here by ruby.
An `if`, wouldn't be affected by this. You could simple do 
```
if app.config.react.components_js.nil?
  app.config.react.components_js = ...
  ...
end
```